source/common/http/http1/balsa_parser.cc

```diff
@@ -0,0 +1,204 @@
+#include "source/common/http/http1/balsa_parser.h"
+
+#include <algorithm>
+#include <cctype>
+#include <cstdint>
+
+#include "source/common/common/assert.h"
+
+#include "absl/strings/match.h"
+
+using ::quiche::BalsaFrameEnums;
+using ::quiche::BalsaHeaders;
+
+namespace Envoy {
+namespace Http {
+namespace Http1 {
+
+namespace {
+constexpr absl::string_view kHttp11Suffix = "HTTP/1.1";
```

Comment:
Can you use Http11String from source/common/http/headers.h, perhaps?

Reply:
Great idea, done.